### ISSUE TITLE
Bug: Transpilation changes algorithm

### ISSUE THREAD

**Issue by medikoo** [MatAtBread/nodent#99]
We had a case as below (here simplified for brevity purposes):

```javascript
var getConstructor = function () {
    return function () {};
};
var  instance = new (getConstructor())()
```

And it got being transpiled to:

```javascript
var getConstructor = function () {
    return function () {};
};
var instance = new getConstructor()();
```

(can be confirmed on [demo page](http://nodent.mailed.me.uk/#var%20getConstructor%20%3D%20function%20()%20%7B%20return%20function%20()%20%7B%7D%3B%20%7D%3B%0A%0Avar%20instance%20%3D%20new%20(getConstructor())()%0A~options~%7B%22mode%22%3A%22promises%22%2C%22promiseType%22%3A%22native%22%2C%22noRuntime%22%3Atrue%2C%22es6target%22%3Atrue%2C%22wrapAwait%22%3Atrue%2C%22spec%22%3Afalse%7D))

The latter has different output, as over there it's `getConstructor` function that's invoked with `new` as _constructor_ and not it's result.

We needed to refactor our code to work around it.

**Comment by matAtWork** [comment]
Yeah, that's a bug. I'll work on a fix today

**Comment by matAtWork** [comment]
Fixed (I hope) in v3.1.3 (now on npmjs). Please re-open if necessary

**Comment by medikoo** [comment]
Thank you!
